editor/editor_interface.cpp

```diff
@@ -263,6 +265,87 @@ void EditorInterface::set_current_feature_profile(const String &p_profile_name)
 	EditorFeatureProfileManager::get_singleton()->set_current_profile(p_profile_name, true);
 }
 
+// Editor dialogs.
+
+void EditorInterface::popup_dialog_scene_tree(const Callable &p_callback, const TypedArray<StringName> &p_valid_types) {
+	// TODO: Should reuse dialog instance instead of creating a fresh one, but need to rework set_valid_types first.
+	if (dialog_scene_tree) {
+		dialog_scene_tree->queue_free();
```

Comment:
I would remove it from the tree first so it doesn't interfere with the new node before getting removed. You should probably also disconnect the signals before any of that.